server/sonar-server/src/main/java/org/sonar/server/platform/web/MasterServletFilter.java

```diff
@@ -123,7 +125,8 @@ public void doFilter(ServletRequest request, ServletResponse response) throws IO
         iterator = filters.iterator();
       }
       if (iterator.hasNext()) {
-        iterator.next().doFilter(request, response, this);
+        // Only ServletFilter should use the ServletRequestWrapper, in order to not use the wrapper in RackFilter
+        iterator.next().doFilter(new ServletRequestWrapper((HttpServletRequest) request), response, this);
```

Comment:
As discussed offline, the ruby code that still uses sessions must be fixed and `getSession()` must always fail.